src/util/getDepth.ts

```diff
@@ -0,0 +1,18 @@
+import { hashContext } from '.'
+import { Context } from '../types'
+import { State } from './initialState'
+import { unroot } from './unroot'
+
+/** Get max depth of a visible context. */
+export const getDepth = (state: State, context: Context): number => {
```

Comment:
We can just use `getDescendants(...).length` to avoid adding an additional function, and it has the benefit of having a single traversal implementation in case the thought structure changes.

Reply:
As per my understanding, `getDescendants` serves a very different purpose from `getDepth` despite some similarities in the traversing logic. The former is a flat map of all the children whereas we need the depth of the longest branch.

Reply:
Oops! You're right. Thanks.